combat-trainer.lic

```diff
@@ -4054,18 +4069,118 @@ class GameState
     @equipment_manager.stow_weapon(weapon_training[skill])
   end
 
-  def wield_offhand_weapon(skill)
+  # Wields a weapon to train the specified skill,
+  # but instead of inferring the weapon name from
+  # `weapon_training:` config, wields the specified
+  # weapon name given.
+  #
+  # Initially designed to support `ambush_stun` attacks
+  # where you need a specific weapon of a particular weight, etc.
+  # so the weapon you want to use may differ from your regular weapon
+  # for the same skill.
+  #
+  # @param weapon_name
+  #   The <adj> + <noun> of the weapon to wield as configured in your gear.
+  # @param skill
+  #   The weapon skill to train.
+  # @param hand
+  #   Which hand to equip the weapon into?
+  #   To equip offhand, specify 'left' or 'offhand'.
+  # @returns name of the wielded weapon
+  def wield_specific_weapon(weapon_name, skill, hand = 'main')
+    echo "wield_specific_weapon: weapon_name=#{weapon_name}, skill=#{skill}, hand=#{hand}" if $debug_mode_ct
+    if is_damaris_weapon?(weapon_name)
+      return wield_damaris_weapon(weapon_name, hand)
+    else
+      if hand =~ /^(left|off)/i
+        @equipment_manager.wield_weapon_offhand?(weapon_name, skill)
+      else
+        @equipment_manager.wield_weapon?(weapon_name, skill)
+      end
+    end
+    return weapon_name
```

Comment:
Why does this return `weapon_name`?

(Same question for the rest of the new functions)

Reply:
Good question, at first I took that piece out because I didn't understand why the method would need to return that value either. Turns out that code that wields a weapon offhand (such as for aiming trainables) uses it to know what weapon to pick back up after you've thrown it.

Without refactoring that code to infer the thrown weapon like we do with inferring ammo to pick up, or introducing a new variable to track your `current_offhand_weapon`, then these wield methods needed to continue returning the weapon name. Both ideas probably worth exploring in future PRs.

![image](https://user-images.githubusercontent.com/68095633/148706022-f9cce1df-721d-4d5e-a761-f9f41ce2bd00.png)

Reply:
Thank you